nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -325,6 +441,859 @@ public Response updateProcessGroup(
+            .collect(Collectors.groupingBy(comp -> comp.getComponentType()));
+
+        final List<AffectedComponentDTO> affectedProcessors = affectedComponentsByType.get(AffectedComponentDTO.COMPONENT_TYPE_PROCESSOR);
+        final List<AffectedComponentDTO> affectedServices = affectedComponentsByType.get(AffectedComponentDTO.COMPONENT_TYPE_CONTROLLER_SERVICE);
+
+
+        if (isReplicateRequest()) {
+            // update the variable registry
+            final VariableRegistryUpdateRequest updateRequest = createVariableRegistryUpdateRequest(groupId);
+            updateRequest.getIdentifyRelevantComponentsStep().setComplete(true);
+            final URI originalUri = getAbsolutePath();
+
+            // Submit the task to be run in the background
+            final Runnable taskWrapper = () -> {
+                try {
+                    updateVariableRegistryReplicated(groupId, originalUri, affectedProcessors, affectedServices, updateRequest, requestEntity);
+                } catch (final Exception e) {
+                    logger.error("Failed to update variable registry", e);
+                    updateRequest.setFailureReason("An unexpected error has occurred: " + e);
+                }
+            };
+
+            variableRegistryThreadPool.submit(taskWrapper);
+
+            final VariableRegistryUpdateRequestEntity responseEntity = new VariableRegistryUpdateRequestEntity();
+            responseEntity.setRequestDto(dtoFactory.createVariableRegistryUpdateRequestDto(updateRequest));
+            responseEntity.getRequestDto().setUri(generateResourceUri("process-groups", groupId, "variable-registry", "update-requests", updateRequest.getRequestId()));
+
+            final URI location = URI.create(responseEntity.getRequestDto().getUri());
+            return Response.status(Status.ACCEPTED).location(location).entity(responseEntity).build();
+        }
+
+
+        final Revision requestRevision = getRevision(requestEntity.getRevision(), requestEntity.getId());
+        return withWriteLock(
+            serviceFacade,
+            requestEntity,
+            requestRevision,
+            lookup -> {
+                final NiFiUser user = NiFiUserUtils.getNiFiUser();
+
+                final Authorizable groupAuthorizable = lookup.getProcessGroup(groupId).getAuthorizable();
+                groupAuthorizable.authorize(authorizer, RequestAction.WRITE, user);
+
+                // For every component that is affected, the user must have READ permissions and WRITE permissions
+                // (because this action requires stopping the component).
+                if (affectedProcessors != null) {
+                    for (final AffectedComponentDTO affectedComponent : affectedProcessors) {
+                        final Authorizable authorizable = lookup.getProcessor(affectedComponent.getComponentId()).getAuthorizable();
+
+                        if (!authorizable.isAuthorized(authorizer, RequestAction.READ, user)) {
```

Comment:
Should use `authorize(...)` here and below. `authorize(...)` will indicate that the user is attempting to access/modify a component. isAuthorize(...)  is meant to simply check if the user is allowed but does not indicate an access attempt. In this context, the user is actually modifying, albeit indirectly, the component.